### PR TITLE
Data collector cleanup

### PR DESCRIPTION
**Pull request by harshjain2**
1. Removed TargetDataCollectorUri as it is not required.
2. Populated TestCaseStartEventArgs.DataCollectionContext.TestCase property. Earlier it was sent as null.

Fixes https://github.com/Microsoft/vstest/issues/1175

### CODE REVIEW

**Review by smadala — APPROVED** [review]

File: src/Microsoft.TestPlatform.ObjectModel/DataCollector/Events/DataCollectionEventArgs.cs [hunk 1]
```diff
@@ -27,17 +27,9 @@ public DataCollectionEventArgs()
...
+        protected DataCollectionEventArgs(DataCollectionContext context)
         {
-        }
-
-        protected DataCollectionEventArgs(DataCollectionContext context, Uri targetDataCollectorUri)
-        {
-            //EqtTrace.FailIf(context == null, "Context should not be null.");
-
             Context = context;
-            TargetDataCollectorUri = targetDataCollectorUri;
```
Comment (singhsarab) [inline src/Microsoft.TestPlatform.ObjectModel/DataCollector/Events/DataCollectionEventArgs.cs]: Keep careful about this change, when you remove the property.

**Review by singhsarab — APPROVED** [review]
with suggestion.

File: src/Microsoft.TestPlatform.ObjectModel/DataCollector/Events/DataCollectionEventArgs.cs (hunk 1 above)
Comment (mayankbansal018) [reply src/Microsoft.TestPlatform.ObjectModel/DataCollector/Events/DataCollectionEventArgs.cs]: @harshjain2 Searched for DataCollectors in devdiv using "TestPlatform.MSTest\ObjectModel\15.0\Microsoft.VisualStudio.TestPlatform.ObjectModel.dll" to compile, they do not seem to be using this property.

https://devdiv.visualstudio.com/DevDiv/_search?type=code&lp=search-Project&text=TestPlatform.MSTest%5CObjectModel%5C15.0&result=DefaultCollection%2FDevDiv%2FVS%2FGBlab%2Fvsumain%2F%2Fsrc%2Fvset%2FQTools%2FExecution%2FAgent%2FPlugins%2FMediaRecorder%2FMediaRecorderV2%2FMediaRecorderV2.csproj&preview=0&filters=ProjectFilters%7BDevDiv%7DRepositoryFilters%7BVS%7DBranchFilters%7Blab%2Fvsumain%7D&_a=search
Remove if from Code, & lets complete the PR

Comment (harshjain2) [reply src/Microsoft.TestPlatform.ObjectModel/DataCollector/Events/DataCollectionEventArgs.cs]: Done.

**Pull request merged**